src/server/lib/experiments.ts

```diff
@@ -0,0 +1,44 @@
+import { Request } from 'express';
+import { parse, stringify } from 'query-string';
+import ms from 'ms';
+
+import { ResponseWithRequestState } from '@/server/models/Express';
+
+const RAN_EXPERIMENTS_COOKIE_NAME = 'ranExperiments';
+
+const getRanExperiments = (req: Request) => {
+  const ranExperimentsValue = req.cookies[RAN_EXPERIMENTS_COOKIE_NAME];
+
+  if (!ranExperimentsValue) return {};
+
+  return parse(ranExperimentsValue);
+};
+
+export const hasExperimentRun = (req: Request, experimentId: string) => {
+  const ranExperiments = getRanExperiments(req);
+  return experimentId in ranExperiments;
+};
+
+export const setExperimentRan = (
+  req: Request,
+  res: ResponseWithRequestState,
+  experimentId: string,
+  ran: boolean,
+) => {
+  const ranExperiments = getRanExperiments(req);
+  let newExperiments = {};
+
+  if (ran) {
+    newExperiments = {
+      ...ranExperiments,
+      [experimentId]: Date.now(),
+    };
+  } else {
+    // Using destructuring to omit property, so extracted variable is unused
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { [experimentId]: omit, ...otherExperiments } = ranExperiments;
+    newExperiments = otherExperiments;
+  }
+  const newValue = stringify(newExperiments);
+  res.cookie(RAN_EXPERIMENTS_COOKIE_NAME, newValue, { maxAge: ms('1yr') });
```

Comment:
In terms of the cookie options I also suggest adding on the `httpOnly` (if not being read on the clientside), `secure`, and `sameSite: "strict"` flags

Reply:
It's not been read on the client currently, but I'm thinking there is a possibility that it might be in the future though, and we wouldn't be able to retroactively change this. So I'd err towards allowing it to be read on the client, unless the information is considered sensitive for some reason (I don't think it is).

Reply:
We've generally gone with `httpOnly` cookies in the past as in general for gateway we've made sure that core functionality isn't reliant on client side code, so avoid reading cookies in the client.

If a client did need a value in a cookie, we'd read it on the server and pass it to the `RequestState` to propagate to the client through one of the properties on there, e.g. `pageData`, and that could be read on the client through the `ClientState`.

Reply:
Ah okay, I see the rationale now. Will change.